### PR TITLE
[DOCS] `s/on-prem/self-managed`

### PR DESCRIPTION
**Pull request by leemthompo**
Tiny substitution

### CODE REVIEW

**Review by jedrazb — COMMENTED** [review]
Should we consider doing `s/on-prem/self-hosted/g`?

https://github.com/search?q=repo%3Aelastic%2Fconnectors-python%20on-prem&type=code there are other places that reference `on-prem` as well

**Comment by seanstory** [comment]
Don't we also use "self managed" instead of "self hosted"?

**Comment by jedrazb** [comment]
@leemthompo should we close this PR since it wasn't merged for a few months?

**Comment by leemthompo** [comment]
@jedrazb this got lost in the mists of time, but I think we can still merge this because the ask was from product to not use "on prem"

File: docs/DOCKER.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ You can download the file manually, or simply run the command below. Make sure t
 curl https://raw.githubusercontent.com/elastic/connectors/main/config.yml --output ~/connectors-config/config.yml
 ```
 
-## 4. Update the configuration file for your [on-prem connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html#build-connector-usage)
+## 4. Update the configuration file for your [self-hosted connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html#build-connector-usage)
```
Comment (leemthompo) [inline docs/DOCKER.md:39]:
```suggestion
## 4. Update the configuration file for your [self-managed connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html#build-connector-usage)
```

**Review by jedrazb — APPROVED** [review]

**Comment by jedrazb** [comment]
Let's gooo!

**Pull request merged**